store/src/java/com/zimbra/cs/lmtpserver/ExternalEmailWarning.java

```diff
@@ -190,9 +193,59 @@ public String getUpdatedContent(String content) {
                     sb.append(getTextHtmlWarning());
                 sb.append(content.substring(end));
                 content = sb.toString();
+            } else if (isIncomingMimeFromGmail(content)) {
```

Comment:
will this fix work if mails are coming from yahoo.com or hotmail.com or any other external domains?

Reply:
so mails coming from the above mails will get the disclaimer or not?

Reply:
if their mime matches existing html regex (Content-Type: text/html.*?\r\n\r\n.*?<body.*?>) then it will get the disclaimer

Reply:
@chitranshu198 you can achieve the same by just making changes in the regex pattern. It will work for other domains too. The below change is not required in that case.